src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,92 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final int RESIZE_MULTIPLIER = 2;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] buckets;
+    private int threshold;
+    private int size;
+
+    public MyHashMap() {
+        buckets = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_LOAD_FACTOR * DEFAULT_CAPACITY);
+    }
 
     @Override
     public void put(K key, V value) {
+        resizeCheck();
+        int index = hash(key);
+        Node<K, V> node = new Node<>(key, value, null);
+        if (buckets[index] == null) {
+            buckets[index] = node;
+            size++;
+        } else {
+            Node<K, V> currentNode = buckets[index];
+            while (currentNode != null) {
+                if (Objects.equals(currentNode.key, key)) {
+                    currentNode.value = value;
+                    return;
+                } else if (currentNode.next == null) {
+                    currentNode.next = node;
+                    size++;
+                    return;
+                }
+                currentNode = currentNode.next;
```

Comment:
Maybe you should put Node in the first position on the list for better performance?